Modules/TPC/src/Tracks.cxx

```diff
@@ -0,0 +1,95 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+///
+/// \file   Tracks.cxx
+/// \author Stefan Heckel, sheckel@cern.ch
+///
+
+// root includes
+#include <TCanvas.h>
+#include <TH1.h>
+#include <TH2.h>
+
+// O2 includes
+#include "Framework/ProcessingContext.h"
+#include "DataFormatsTPC/TrackTPC.h"
+
+// QC includes
+#include "QualityControl/QcInfoLogger.h"
+#include "TPC/Tracks.h"
+
+namespace o2::quality_control_modules::tpc
+{
+
+Tracks::Tracks() : TaskInterface() {}
+
+Tracks::~Tracks()
+{
+}
+
+void Tracks::initialize(o2::framework::InitContext& /*ctx*/)
+{
+  QcInfoLogger::GetInstance() << "initialize TPC Tracks QC task" << AliceO2::InfoLogger::InfoLogger::endm;
+
+  mQCTracks.initializeHistograms();
+
+  for (auto& hist : mQCTracks.getHistograms1D()) {
+    getObjectsManager()->startPublishing(&hist);
+    getObjectsManager()->addMetadata(hist.GetName(), "custom", "34");
+  }
+
+  for (auto& hist2 : mQCTracks.getHistograms2D()) {
+    getObjectsManager()->startPublishing(&hist2);
+    getObjectsManager()->addMetadata(hist2.GetName(), "custom", "42");
+  }
+}
+
+void Tracks::startOfActivity(Activity& /*activity*/)
+{
+  QcInfoLogger::GetInstance() << "startOfActivity" << AliceO2::InfoLogger::InfoLogger::endm;
+  mQCTracks.resetHistograms();
+}
+
+void Tracks::startOfCycle()
+{
+  QcInfoLogger::GetInstance() << "startOfCycle" << AliceO2::InfoLogger::InfoLogger::endm;
+}
+
+void Tracks::monitorData(o2::framework::ProcessingContext& ctx)
+{
+  using TrackType = std::vector<o2::tpc::TrackTPC>;
+  auto tracks = ctx.inputs().get<TrackType>("inputTracks");
```

Comment:
Here, it would be better to change it to:
```c++
using TracksType = gsl::span<o2::tpc::TrackTPC>;
const auto tracks = ctx.inputs().get<TracksType>("inputTracks");
```
and add
```c++
#include <gsl/span>
```
See discussion in https://alice.its.cern.ch/jira/browse/O2-1017.

@tklemenz : Please also consider this and the comment below for the PID qc task.